system/lib/pthread/library_pthread_stub.c

```diff
@@ -389,13 +390,22 @@ void __acquire_ptc() {}
 
 void __release_ptc() {}
 
-// When pthreads is not enabled, we can't use the Atomics futex api to do
-// proper sleeps, so simulate a busy spin wait loop instead.
-void emscripten_thread_sleep(double msecs) {
+// busy spin wait sleep
+static void busy_sleep(double msecs) {
   double start = emscripten_get_now();
   double now = start;
   do {
     _emscripten_yield(now);
```

Comment:
One concern I have is that `emscripten_yield` here used to run `setitimer` callbacks, proxying.h callbacks and so on while the main thread is "sleeping".

Now all of them will be blocked until the sleep timeout has ended. Can that cause performance issues or maybe even deadlocks?

Reply:
`futex_wait_main_browser_thread` seems semantically very similar to this kind of sleeping, and it specifically has logic to handle all those events in the middle of atomic sleep, with following comment:

https://github.com/emscripten-core/emscripten/blob/9bdb310b89472a0f4d64f36e4a79273d8dc7fa98/system/lib/pthread/emscripten_futex_wait.c#L56-L57

Wouldn't same concerns apply here?

Reply:
Another data point - `emscripten_thread_sleep` has following documentation:

https://github.com/emscripten-core/emscripten/blob/9bdb310b89472a0f4d64f36e4a79273d8dc7fa98/system/include/emscripten/threading.h#L78-L85

It's surprising that it's not caught by tests, but it seems like this change will break that contract about handling proxying events?

Reply:
I guess proxying callbacks don't matter since this is used only for build without `-pthread` (right?), but `setitimer` events probably will be broken?